packages/cli/src/routeGeneration/templates/express.hbs

```diff
@@ -72,7 +72,14 @@ export function RegisterRoutes(app: express.Express) {
             }
 
             {{#if ../../iocModule}}
-            const controller: any = iocContainer.get<{{../name}}>({{../name}});
+            function isFunction(obj: any): obj is Function { return typeof obj === 'function'; }
+            interface Container<T> { get<T>(type: any): any; }
+
+            let container: any = iocContainer;
+            if (isFunction(iocContainer)) {
+                container = iocContainer(request);
+            }
+            const controller: any = (container as Container<{{../name}}> ).get<{{../name}}>({{../name}});
```

Comment:
Looking back at the issue, the suggested API was container.get(name, request), not container(request).get(name).
What's the rationale for the change here>

Reply:
Because the change is about fetching a container, it made more sense to be `container(request).get(name)`, as the `container()` method fetches the container, which we then call `.get()` on.

`container.get(name, request)` also didn't seem workable and would be a breaking change.